spec/batch.spec.js

```diff
@@ -89,6 +89,28 @@ describe('batch', () => {
     expect(internalURL).toEqual('/classes/Object');
   });
 
+  it('should return the proper url with bad url provided', () => {
```

Comment:
These tests are added because the design of `batch.makeBatchRoutingPathFunction` attempts to handle this, but these two added test fail with the `url.parse` implementation of the code.